### PR TITLE
fix: Check out jx-requirements.yml from cluster repo upon sparse checkout

### PR DESCRIPTION
**Pull request by GavinPJK**
Requirements config file is needed `jx-requirements.yml` as well as `.jx/gitops/source-config.yaml` - add the former when doing a sparse checkout.

Note: tests are failing as only the "clone" step is run: https://github.com/jenkins-x-plugins/jx-gitops/blob/main/pkg/fakerunners/git_cloner.go

### CODE REVIEW

File: pkg/variablefinders/requirements.go [hunk 1]
```diff
@@ -95,7 +95,11 @@ func GetSettings(g gitclient.Interface, jxClient jxc.Interface, ns, dir, owner,
 			return nil, "", errors.New("failed to find a dev environment source url on development environment resource")
 		}
 	}
-	clusterDir, err := requirements.PartialCloneClusterRepo(g, gitURL, true, sourceconfigs.SourceConfigFile)
+	combinedConfigFiles := []string{"/" + jxcore.RequirementsConfigFileName, sourceconfigs.SourceConfigFile}
+	if len(combinedConfigFiles) == 0 {
+		return nil, "", errors.New("combined config files are empty")
+	}
```
Comment (Skisocks) [inline pkg/variablefinders/requirements.go:101]: This is redundant as it's always populated

**Review by Skisocks — CHANGES_REQUESTED** [review]

File: pkg/variablefinders/requirements.go (hunk 1 above)
Comment (GavinPJK) [reply pkg/variablefinders/requirements.go]: Done.

**Review by Skisocks — APPROVED** [review]

**Pull request merged**